native-src/android/app/src/main/java/com/telerik/localnotifications/Builder.java

```diff
@@ -0,0 +1,269 @@
+package com.telerik.localnotifications;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.support.v4.app.NotificationCompat;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+
+import static android.app.PendingIntent.FLAG_UPDATE_CURRENT;
+
+public final class Builder {
+
+    private static final String TAG = "Builder";
+    private static final String DEFAULT_CHANNEL = "Notifications";
+
+    // To generate unique request codes:
+    private static final Random RANDOM = new Random();
```

Comment:
What kind of uniqueness do you need here?
Maybe it is better to use UUID

Reply:
If you had an activity receiving Intents from 2 other activities, you would use this `requestId` to tell which one did it come from, it doesn't need to be unique.

Anyway, in this case, this is not used, because we had the notification ID as an extra, so I'm not really sure this matters (maybe it could just be 0) or maybe we could use the notification ID as `requestId` and get rid of the extra.

Reply:
If we can, better to get rid of it

Reply:
Working on this now. Will call you after the daily to discuss something with you.

Reply:
Updated, it should be ok now, at the end I implemented the generation of the notification ID as `Date.now() + (10e6 * Math.random() | 0 )`. The notification ID is now used as request ID as well for `PendingInten` and other extras have been removed in favour of `Intent#setAction`.